### DISCUSSION TITLE
Bigger upload  chunk size  slower read single  file  from s3zip

### DISCUSSION THREAD

**Discussion by CoderXionghs** [minio/minio#16152] [Q&A] [answered]
I use  s3zip feature to package  many  small  files  and  then  read  single  small  file  from zip。when i  test the  read  performance，i found  the  upload chunk size  will affect  the  read single small  file speed，the  bigger  size the slower speed。
       for example：a zip file with  4GB ，contains 10000 small files，no compressed。when chunkSize=10MB,single small file  read average speed=4MB/s，when chunkSize=1024M，the  read  speed only  100KB/s。

So  can  any people explain  why? and  how set a best  upload  chunkSize?

My test  environment:
  Minio-version: 2022-11-08T05:27:07Z
  distribution deploy: three  node，every node two  device，arm64 cpu

**Comment by harshavardhana** [comment, marked as answer]
This is normal right, s3zip is meant for small files. 

Why would you use it for bigger files ?

**Reply by klauspost** [reply]
It should work perfectly fine for bigger files as well.

**Reply by CoderXionghs** [reply]
> This is normal right, s3zip is meant for small files.
> 
> Why would you use it for bigger files ?

I am not  package  big files to zip，i packaged  many  small files to zip，the  final  zip  size is 4GB，the small file's average size is 400KB

**Comment by klauspost** [comment]
@CoderXionghs What do you mean "chunk size"? Do you mean part size?

**Reply by CoderXionghs** [reply]
yes，the part size

**Reply by CoderXionghs** [reply]
> @CoderXionghs What do you mean "chunk size"? Do you mean part size?

yes，i mean the  part  size

**Reply by harshavardhana** [reply]
1024MiB part size will be slow because your network is not capable of handling such a large stream. Upload in smaller chunks.

**Reply by CoderXionghs** [reply]
> 1024MiB part size will be slow because your network is not capable of handling such a large stream. Upload in smaller chunks.

do you  mean，when  read  a  small file inside s3zip，minio  need  read the whole part.x？ i read the  source  code and  it shows: minio will  find the offset of the small file,only  need to  read  the  compressed size content.

**Reply by klauspost** [reply]
It should be able to skip directly to the content. I will test.

**Reply by klauspost** [reply]
I found a bug, where index is not written to the object. Adding a benchmark to [warp](https://github.com/minio/warp) and sending a fix.

**Reply by klauspost** [reply]
Also found the cause of the part-dependent slowdown.

**Reply by klauspost** [reply]
@CoderXionghs https://github.com/minio/minio/pull/16161

**Reply by CoderXionghs** [reply]
@klauspost  thanks for your quicky replay
